lib/oauth2/response.rb

```diff
@@ -78,12 +78,21 @@ def parsed
             parser.call(body, response)
           end
         end
+
+      if @parsed.is_a?(Hash)
```

Comment:
Would like to remove the internal handling, and use a tool that gives us more coverage, such as the ability to snake case the hashes inside JSON **arrays** as well.

[plissken](https://github.com/futurechimp/plissken) looks like a good candidate.